### PR TITLE
Java errorreporting migration

### PR DESCRIPTION
**Pull request by minherz**
Migrate snippets from [java-errorreporting][] repo.

[java-errorreporting]: https://github.com/googleapis/java-errorreporting/

### CODE REVIEW

File: errorreporting/pom.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.google.cloud</groupId>
+  <artifactId>java-docs-samples-errorreporting</artifactId>
+  <version>0.0.1-SNAPSHOT</version><!-- This artifact should not be released -->
+  <packaging>pom</packaging>
+  <name>Google Cloud Error Reporting Samples Parent</name>
+  <url>https://github.com/GoogleCloudPlatform/java-docs-samples/errorreporting</url>
```
Comment (Shabirmean) [inline errorreporting/pom.xml:9]: The url needs the `/tree/main/` part for it to work!

Comment (minherz) [reply errorreporting/pom.xml:9]: Thanx!

File: errorreporting/pom.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (Shabirmean) [inline errorreporting/pom.xml:1]: I think we don't need this pom. This file must be removed!

**Review by averikitsch — APPROVED** [review]

File: errorreporting/pom.xml (hunk 2 above)
Comment (Shabirmean) [reply errorreporting/pom.xml:1]: CC: @averikitsch to confirm

Comment (minherz) [reply errorreporting/pom.xml:1]: Should I roll everything one directory up then?

**Review by Shabirmean — APPROVED** [review]

**Pull request merged**